client/src/components/forms/LoginForm.js

```diff
@@ -0,0 +1,82 @@
+import { observer } from "mobx-react-lite";
+import { useLocation, useNavigate } from "react-router-dom";
+import { useContext, useState } from "react";
+import { Context } from "../..";
+
+import { EXPLORE_PAGE, HOME_PAGE } from "../../utils/constans";
+import { authentication } from "../../http/userApi";
+import ButtonClose from "../buttons/ButtonClose";
+import LoginPasswordForm from "./LoginPasswordForm";
+import LoginEmailForm from "./LoginEmailForm";
+import Logo from "../common/Logo";
+
+import "./loginAndRegistretionForm.css";
+
+const LoginForm = observer(() => {
+  const { usersStore } = useContext(Context);
+  const navigate = useNavigate();
+  const [showFormPassword, setShowFormPassword] = useState(false);
+  const [showFormEmail, setShowFormEmail] = useState(true);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const location = useLocation().pathname;
+
+  const showPasswordForm = () => {
+    if (email) {
+      setShowFormPassword({ showFormPassword: true });
+    }
+  };
+
+  const signIn = async () => {
+    try {
+      if (email && password) {
+        const userProfile = await authentication(email, password);
+
+        usersStore.setUser(userProfile);
+        usersStore.setAuth(true);
```

Comment:
Try to follow the "Single level of abstraction rule" here:

```suggestion
        const authenticationResult = await authClient.authenticate(email, password);

        usersStore.setUserAuth(authenticationResult.user);
        localAuthClient.setAuthToken(authenticationResult.token);
        
```